spyder_kernels/customize/code_runner.py

```diff
@@ -0,0 +1,593 @@
+        filename = canonic_filename
+
+        with NamespaceManager(
+            self.shell,
+            filename,
+            current_namespace=current_namespace,
+            file_code=file_code,
+            context_locals=context_locals,
+            context_globals=context_globals,
+        ) as (ns_globals, ns_locals):
+            sys.argv = [filename]
+            if args is not None:
+                # args are a sting in a string
+                for arg in shlex.split(args):
+                    sys.argv.append(arg)
+
+            if "multiprocessing" in sys.modules:
+                # See https://github.com/spyder-ide/spyder/issues/16696
+                try:
+                    sys.modules["__mp_main__"] = sys.modules["__main__"]
+                except Exception:
+                    pass
+
+            if wdir is not None:
+                if wdir is True:
+                    # True means use file dir
+                    wdir = os.path.dirname(filename)
+                if os.path.isdir(wdir):
+                    os.chdir(wdir)
+                    # See https://github.com/spyder-ide/spyder/issues/13632
+                    if "multiprocessing.process" in sys.modules:
+                        try:
+                            import multiprocessing.process
+
+                            multiprocessing.process.ORIGINAL_DIR = os.path.abspath(wdir)
+                        except Exception:
+                            pass
+                else:
+                    print("Working directory {} doesn't exist.\n".format(wdir))
+
+            try:
+                if __umr__.has_cython:
+                    # Cython files
+                    with io.open(filename, encoding="utf-8") as f:
+                        self.shell.run_cell_magic("cython", "", f.read())
+                else:
+                    self._exec_code(
+                        file_code,
+                        filename,
+                        ns_globals,
+                        ns_locals,
+                        post_mortem=post_mortem,
+                        exec_fun=exec_fun,
+                        capture_last_expression=False,
+                        global_warning=not current_namespace,
+                    )
+            finally:
+                sys.argv = [""]
+
+    def _exec_cell(
+        self,
+        cell_id,
+        filename=None,
+        post_mortem=False,
+        exec_fun=None,
+        canonic_filename=None,
+        context_locals=None,
+        context_globals=None,
+    ):
+        """
+        Execute a code cell.
+        """
+        try:
+            # Get code from spyder
+            cell_code = frontend_request(blocking=True).run_cell(cell_id, filename)
+        except Exception:
+            print(
+                "This command failed to be executed because an error occurred"
+                " while trying to get the cell code from Spyder's"
+                " editor. The error was:\n\n"
```

Comment:
```suggestion
                "This command failed to be executed because an error occurred "
                "while trying to get the cell code from Spyder's editor."
                "The error was:\n\n"
```